modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionsEvictManager.java

```diff
@@ -718,19 +719,21 @@ public EvictReason reason() {
         /**
          * Submits the task for execution.
          *
-         * @return {@code True} if the task was submitted for execution.
+         * @return {@link Future} if the task was submitted for execution.
          */
-        public boolean start() {
+        public Future<?> start() {
```

Comment:
I don't think that you need to change the contract. You always can use `finishFuture` from this task as follows:
```
    if (task.run())
        task.finishFuture().get()
```

Reply:
Fixed